### PR TITLE
feat(widget-v2): support experimental features

### PR DESCRIPTION
**Pull request by samobasquiat**
# Summary
In this task, we were supposed to support experimental features. These Items are set in the dapp & storage and in the widget, we read apiUrl or experimental best route from config. 

# How did you test this change?
For testing, we can manually set config values for experimentalFeatures & apiUrl to change the request body of best route & apiUrl in our fetches.

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: widget/embedded/src/constants/index.ts [hunk 1]
```diff
@@ -1 +1,5 @@
+import { getExperimentalBaseUrl } from '../utils/quote';
+
 export const RANGO_PUBLIC_API_KEY = 'c6381a79-2817-4602-83bf-6a641a409e32';
+export const DEFAULT_BASE_URL = 'https://api.rango.exchange';
+export const RANGO_BASE_URL = getExperimentalBaseUrl() || DEFAULT_BASE_URL;
```
Comment (mikasackermn) [inline widget/embedded/src/constants/index.ts:5]: It seems that changing the base URL has no effect. please check.

File: widget/embedded/src/Widget.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import React, { useContext, useEffect, useMemo, useState } from 'react';
 import { AppRouter } from './components/AppRouter';
 import { AppRoutes } from './components/AppRoutes';
 import { WidgetEvents } from './components/WidgetEvents';
+import { RANGO_BASE_URL } from './constants';
 import { globalFont } from './globalStyles';
 import { useLanguage } from './hooks/useLanguage';
 import { useTheme } from './hooks/useTheme';
```
Comment (mikasackermn) [inline widget/embedded/src/Widget.tsx:14]: needReloadTheApp is one of the props related to ExperimentalFeatures located in the Dapp and if true, the app should be reloaded if the URL changes, and if false, no reload is required (not implemented in the widget).

File: widget/embedded/src/QueueManager.tsx [hunk 3]
```diff
@@ -43,7 +44,7 @@ function QueueManager(props: PropsWithChildren) {
     })),
   };
 
-  const switchNetwork = (wallet: WalletType, network: Network) => {
+  const switchNetwork = async (wallet: WalletType, network: Network) => {
```
[the remark this answers is not in the document]
Comment (samobasquiat) [reply widget/embedded/src/QueueManager.tsx:47]: It was added by Eslint. We have to.

File: widget/embedded/src/Widget.tsx (hunk 2 above)
Comment (samobasquiat) [reply widget/embedded/src/Widget.tsx:14]: yeah but I think It should be handled in dapp not here. In dapp we reload the page and that's why the widget is reloaded as well.

File: widget/embedded/src/constants/index.ts (hunk 1 above)
Comment (samobasquiat) [reply widget/embedded/src/constants/index.ts:5]: It was my mistake. fixed.

**Review by mikasackermn — APPROVED** [review]

**Comment by yeager-eren** [comment]
I would suggest to consider it as a config and not a separate thing and implement it using local storage directly.

The final config can be like this:

```
config = {
   apiUrl?: string 
   experimental?: {
        routing?: boolean
   }
}
```

File: widget/embedded/src/utils/configs.ts [hunk 4]
```diff
@@ -2,14 +2,15 @@ import { RANGO_PUBLIC_API_KEY } from '../constants';
```
Comment (yeager-eren) [inline widget/embedded/src/utils/configs.ts:2]: I think this file changes should be reverted.

File: widget/embedded/src/services/httpService.ts [hunk 5]
```diff
@@ -8,6 +8,6 @@ export const httpService = () => {
   if (rango) {
     return rango;
   }
-  rango = new RangoClient(getConfig('API_KEY'));
+  rango = new RangoClient(getConfig('API_KEY'), getConfig('BASE_URL'));
```
Comment (yeager-eren) [inline widget/embedded/src/services/httpService.ts:11]: I think this one should be reverted as well.

File: widget/embedded/src/utils/configs.ts (hunk 4 above)
Comment (yeager-eren) [reply widget/embedded/src/utils/configs.ts:2]: My bad. resolve it.

**Pull request merged**